certbot-apache/certbot_apache/augeasparser.py

```diff
@@ -67,8 +67,30 @@ def __eq__(self, other):
         return False
 
     def set_parameters(self, parameters):
-        """Sets the parameters for DirectiveNode"""
-        self.parameters = parameters
+        """
+        Sets parameters of a DirectiveNode or BlockNode object.
+
+        :param list parameters: List of all parameters for the node to set.
+        """
+        orig_params = self._aug_get_params(self.metadata["augeaspath"])
+
+        # Clear out old parameters
+        for _ in orig_params:
+            # When the first parameter is removed, the indices get updated
+            param_path = "{}/arg[1]".format(self.metadata["augeaspath"])
+            self.parser.aug.remove(param_path)
+        # Insert new ones
+        for pi, _ in enumerate(parameters):
+            param_path = "{}/arg[{}]".format(self.metadata["augeaspath"], pi+1)
+            self.parser.aug.set(param_path, parameters[pi])
+
+        self.parameters = tuple(parameters)
+
+    def _aug_get_params(self, path):
+        """Helper function to get parameters for DirectiveNodes and BlockNodes"""
+
+        arg_paths = self.parser.aug.match(path + "/arg")
```

Comment:
Really? They just use the same syntax? Maybe someone who knows augeas syntax should pop in for a quick look... @bmw mind double-checking the augeas bits?

Reply:
Sure. I don't think I'll get to this today, but I can take a look tomorrow or early next week at the latest.